File: Iteration2/code/CSE3063GRP12/src/core/models/concretes/Advisor.java
package core.models.concretes;

import java.util.ArrayList;

public class Advisor extends Lecturer {
    private ArrayList<String> listOfStudentIds;

    public Advisor(String id, String firstName, String lastName, String userName,
            String password, ArrayList<Course> listOfCoursesGiven,
            ArrayList<String> listOfStudentIds) {
        super(id, firstName, lastName, userName, password, listOfCoursesGiven);
        this.listOfStudentIds = listOfStudentIds;
    }

    public void setListOfStudentIds(ArrayList<String> listOfStudents) {
        this.listOfStudentIds = listOfStudents;
    }

    public ArrayList<String> getListOfStudentIds() {
        return listOfStudentIds;
    }
}

File: Iteration3/code/CSE3063GRP12/src/core/models/concretes/Advisor.py
from core.models.concretes.Lecturer import Lecturer

class Advisor(Lecturer):
    def __init__(self, id=None, firstName=None, lastName=None, userName=None, password=None, listOfCoursesGiven=None, listOfStudentIds=None):
        super().__init__(id, firstName, lastName, userName, password, listOfCoursesGiven)
        self.listOfStudentIds = listOfStudentIds if listOfStudentIds else []
    
    def to_dict(self):
        return {
            "id": self.id,
            "firstName": self.firstName,
            "lastName": self.lastName,
            "userName": self.userName,
            "password": self.password,
            "listOfCoursesGiven": [course.to_dict() if hasattr(course, 'to_dict') else course for course in self.listOfCoursesGiven],
            "listOfStudentIds": self.listOfStudentIds
        }